module/zfs/arc.c

```diff
@@ -3677,6 +3698,76 @@ arc_alloc_raw_buf(spa_t *spa, void *tag, uint64_t dsobj, boolean_t byteorder,
 	return (buf);
 }
 
+static void
+l2arc_hdr_arcstats_update(arc_buf_hdr_t *hdr, boolean_t incr,
+    boolean_t state_only)
+{
+	l2arc_buf_hdr_t *l2hdr = &hdr->b_l2hdr;
+	l2arc_dev_t *dev = l2hdr->b_dev;
+	uint64_t lsize = HDR_GET_LSIZE(hdr);
+	uint64_t psize = HDR_GET_PSIZE(hdr);
+	uint64_t asize = vdev_psize_to_asize(dev->l2ad_vdev, psize);
+	arc_buf_contents_t type = hdr->b_type;
+	int64_t lsize_s;
+	int64_t psize_s;
+	int64_t asize_s;
+
+	if (!incr) {
```

Comment:
Might as well flip the order of these to avoid the negation.

Reply:
Could you elaborate? Even if I flip the conditional, I would not avoid the negation.

Reply:
I just mean the negation in the condition:
```C
if (incr) {
	lsize_s = lsize;
 	psize_s = psize;
 	asize_s = asize;
} else {
	lsize_s = -lsize;
 	psize_s = -psize;
 	asize_s = -asize;
}
```